packages/java/endpoint/src/main/java/com/vaadin/hilla/route/RouteUnifyingIndexHtmlRequestListener.java

```diff
@@ -142,18 +140,21 @@ protected Map<String, AvailableViewInfo> collectClientViews(
 
         var clientViews = new HashMap<String, AvailableViewInfo>();
         clientRouteRegistry.getAllRoutes().entrySet().stream()
-                .filter(clientViewConfigEntry -> routeUtil.isRouteAllowed(
-                        isUserInRole, isUserAuthenticated,
-                        clientViewConfigEntry.getValue()))
-                .forEach(clientViewConfigEntry -> {
+                .filter(clientViewConfigEntry -> {
+                    var config = clientViewConfigEntry.getValue();
+                    return routeUtil.isRouteAllowed(isUserInRole,
+                            isUserAuthenticated, config)
+                            && (config.getRouteParameters() == null
+                                    || config.getRouteParameters().isEmpty());
```

Comment:
I would do this in the following order/style:
1. map clientViewConfigEntry -> clientViewConfigEntry.getValue()
2. filter those satisfying `(config.getRouteParameters() == null || config.getRouteParameters().isEmpty())`
3. filter those satisfying `routeUtil.isRouteAllowed(isUserInRole, isUserAuthenticated, config)`

Reply:
I would also replace the `.forEach` part with 
```java
.collect(Collectors.toMap(AvailableViewInfo::route, Function.identity()))
```